### ISSUE TITLE
Unit tests crash, flooding output with long stack trace

### ISSUE THREAD

**Issue by fedde-s** [cBioPortal/cbioportal-frontend#910]
When I run the test suite on the `rc` or `master` development branches, this test crashes with a gigantic stack trace, scrolling all the other test results off the screen:

```
  QueryStoreUtils
    nonMolecularProfileParams
      ✗ returns normalized query for gene_list parameter
```

**UPDATE:** this is not the only crashing test, all tests in the `StructureVisualizer3D` test case seem to crash in the same way.

**Comment by fedde-s** [comment]
The error message is ‘Maximum call stack size exceeded.’ and the stack trace is rather repetitive:
```
        common-dist/common.bundle.js:8044:40
        common-dist/common.bundle.js:8044:40
        common-dist/common.bundle.js:8044:40
[repeated over 940 times]
```

**Comment by adamabeshouse** [comment]
@fedde-s are you on the latest version of master?

**Comment by fedde-s** [comment]
Yes, and of `rc`. The first commit I've been able to reproduce this issue with seems to be 353e74056c3ac89a0695930b245349d8b97099d4.

**Comment by fedde-s** [comment]
I've asked a few others to reproduce it by testing master on their systems to rule out any fuzzy details of my installation, and it failed at the same test for them.

**Comment by adamabeshouse** [comment]
when I run that test alone `npm run test -- --grep=QueryStoreUtils` , I dont see any stack trace. When I read the test it's quite inoffensive and I would be surprised if it caused this. That's why I'm confused. My idea is there must be some other test that's run earlier whose infinite loop doesn't show up until that test, just by virtue of timing. I'm looking into running tests in random order to get more clues/prove this theory

**Comment by fedde-s** [comment]
I can confirm that the command you mentioned succeeds fine on my side as well. Here's the log of the last few testcases I see listed before the failing test, when I leave out the `--grep` option and parameter:

```
[…]
  DataTypePrioritySelector
    profileAvailability
      ✓ returns correct profile availability result in case of zero profiles
      ✓ returns correct profile availability result in case of one profile
      ✓ returns correct profile availability result in case of two profiles
      ✓ returns correct profile availability result in case of several profiles
    radioButtons
      ✓ shows the right buttons when theres mutation and cna profiles available
      ✓ shows the right buttons when theres just mutation profiles available
      ✓ shows the right buttons when theres just cna profiles available

  QueryAndDownloadTabs
    ✓ Hides download tab if prop showDownloadTab is false

  QueryStoreUtils
    nonMolecularProfileParams
      ✗ returns normalized query for gene_list parameter
        Maximum call stack size exceeded.
        common-dist/common.bundle.js:1:108590
        common-dist/common.bundle.js:1:108590
        common-dist/common.bundle.js:1:108590
[…]
```

**Comment by adamabeshouse** [comment]
I saw those too, after setting my terminal to infinite scrollback :D but none of those seem to trigger the problem, even if I add dummy tests that are just 1500 ms timeouts to stall until the stack overflow can appear. The other weird thing is the tests arent failing on Travis

**Comment by fedde-s** [comment]
Taking another look, it's not limited to the one test I listed. All the tests in the `StructureVisualizer3D` case fail with a very similar stack trace, starting with ‘`StructureVisualizer3D` properly initializes the 3D visualizer’).

**Comment by alisman** [comment]
i will take a look first thing.

**Comment by alisman** [comment]
I'm pretty sure this has something to do with MobX.  I think we are setting that observable properties on something which causes major reaction or something.  Still working on it.

**Comment by alisman** [comment]
Ok, it seems to be this test:
"initializes selection for given tiers" in ResultsViewPageStoreUtils.  It's something to do with the Observable.map usage.  Possible deepEqual applied to observable map.  Not sure.  When I comment that one test out, we're good.  @fedde-s @adamabeshouse can you confirm this?   I suspect it passes on travis because this isn't a logic issue, it crashes because of max callstack exceeded, which is going to be environment dependent.   Once others confirm that it's this test, we can spend more time to figure out exactly what the problem is and and how to avoid it.  Other more urgent problems right now : )

**Comment by alisman** [comment]
We've merged PR to master which skips these these tests.  Will merge to rc later today.

**Comment by fedde-s** [comment]
That solves the issue as reported above; the potential new blind spot in the test coverage is an issue of another type.

> Type 'ITechnicalDebt' is not assignable to type 'IBugReport'. Property 'howToReproduce' is missing in type 'ITechnicalDebt.

**Comment by adamabeshouse** [comment]
Lets keep this open so we eventually figure it out.

**Comment by fedde-s** [comment]
Sure!

**Comment by fedde-s** [comment]
The issue as described above does not occur anymore.
